sql/plsql/PlSqlParser.g4

```diff
@@ -3090,19 +3309,61 @@ end_time_column
     ;
 
 column_definition
-    : column_name (datatype | type_name)
-         SORT? (VISIBLE | INVISIBLE)? (DEFAULT expression)? (ENCRYPT (USING  CHAR_STRING)? (IDENTIFIED BY regular_id)? CHAR_STRING? (NO? SALT)? )?  (inline_constraint* | inline_ref_constraint)
+    : column_name
+         ( (datatype | regular_id) (COLLATE column_collation_name)?)?
+         SORT?
+         (VISIBLE | INVISIBLE)?
+         (DEFAULT (ON NULL_)? expression | identity_clause)?
+         (ENCRYPT encryption_spec)?
+         (inline_constraint+ | inline_ref_constraint)?
+    ;
+
+column_collation_name
+    : id_expression
+    ;
+
+identity_clause
+    : GENERATED (ALWAYS | BY DEFAULT (ON NULL_)?)? AS IDENTITY ('(' identity_options* ')')?
+    ;
+
+identity_options
+    : START WITH (numeric | LIMIT VALUE)
+    | INCREMENT BY numeric
+    | MAXVALUE numeric
+    | NOMAXVALUE
+    | MINVALUE numeric
+    | NOMINVALUE
+    | CYCLE
+    | NOCYCLE
+    | CACHE numeric
+    | NOCACHE
+    | ORDER
+    | NOORDER
     ;
 
 virtual_column_definition
-    : column_name datatype? autogenerated_sequence_definition?
-        VIRTUAL? inline_constraint*
+    : column_name (datatype COLLATE column_collation_name)?
+        (VISIBLE | INVISIBLE)?
+        autogenerated_sequence_definition?
+        VIRTUAL?
+        evaluation_edition_clause?
+        unusable_edition_clause
```

Comment:
I suggest changing it to optional:

```suggestion
        unusable_edition_clause?
```

And making `unusable_edition_clause` not empty.